### PR TITLE
Signup page

### LINKED ISSUE #11: Create signup box component
- First name entry
- Last name entry
- Email entry
- Password entry
- Confirm password
- Submit entries
- Cancel sign up

### PR DESCRIPTION
**Pull request by max-gurr**
- Add SignUp page
- Create SignUp component with text fields and signup button
- TextField global component
- Link textfields with updateable placeholder values for user-feedback

### CODE REVIEW

**Review by franc-hoon — COMMENTED** [review]
Does the Sign Up page have the image on the left? I provided all the page images in the assets/images folder.
It'll also be great if you could somehow include a 'Sign up with Google' button (which can be re-used in the login page too).
Other than that, seems good!

**Comment by max-gurr** [comment]
@M-Stable I've added the bg image. Google signup would be an extra feature, if we want to include it can be a future feature

**Review by franc-hoon — APPROVED** [review]

File: src/components/global/TextInput.jsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+function TextInput(props) {
+  const { label, textValue, placeholderValue, onChangeHandler } = props;
+  
+  return (
+    <div className="text-input">
+      {label && 
+        <label className="text-input--label">
```
Comment (yuno99825) [inline src/components/global/TextInput.jsx]: we've been using '--' in class names for conditions/states, for example button--disabled. Maybe consider changing this to a single -

File: src/pages/signUpView.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    const [ email, setEmail ] = useState("");
+    const [ password, setPassword ] = useState("");
+    const [ firstNameInfoText, setFirstNameInfoText ] = useState("");
+    const [ lastNameInfoText, setLastNameInfoText ] = useState("");
+    const [ emailInfoText, setEmailInfoText ] = useState("");
+    const [ passwordInfoText, setPasswordInfoText ] = useState("");
+
+    const signUp = () => {
+        // For debugging
+        console.log(`First name: ${firstName} \nLast name: ${lastName} \nEmail: ${email} \nPassword: ${password}`);
```
Comment (yuno99825) [inline src/pages/signUpView.js:19]: Remove this console.log

File: src/pages/signUpView.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+        // Reset variables
+        setFirstName("");
+        setLastName("");
+        setEmail("");
+        setPassword("");
+    }
+
+    return (
+        <div className="sign-up--root">
```
Comment (yuno99825) [inline src/pages/signUpView.js]: Classnames, same as above.

File: src/stylesheets/signup.scss [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+            margin-bottom: 30px;
+        }
+
+        .button-text {
+            font-size: 0.8em;
+        }
+    }
+
+    &--bottom-text {
+        color: #818181;
```
Comment (yuno99825) [inline src/stylesheets/signup.scss]: consider using a variable here.

**Review by yuno99825 — COMMENTED** [review]
Looks good, left a few comments, mostly to do with css class name conventions.

File: src/pages/signUpView.js (hunk 2 above)
Comment (max-gurr) [reply src/pages/signUpView.js:19]: It'll be removed before it's merged into main. Jesse and I need to collate our work on this branch before we implement the authorisation stuff

File: src/pages/signUpView.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+            <div className="sign-up--container">
```
Comment (franc-hoon) [inline src/pages/signUpView.js]: Hmm, I wonder if it's actually better to use a Form here as convention

Comment (max-gurr) [reply src/pages/signUpView.js]: I thought about it, but after doing some research it doesn't seem that forms are that necessary anymore. Their main use is if you're sending the data straight to a server

**Review by yuno99825 — APPROVED** [review]

**Pull request merged**